### DISCUSSION TITLE
Question about ReactDOMServer.renderToStaticMarkup

### DISCUSSION THREAD

**Discussion by ashiq-prospero** [igordanchenko/react-photo-album#168] [Q&A] [answered]
There is some issue with ReactDOMServer.renderToStaticMarkup, with the below code it renders the main container i.e react-photo-album--rows but its empty, images are not rendered

```
ReactDOMServer.renderToStaticMarkup(<PhotoAlbum
  photos={photos}
  {...config}
/>);
```

But when I add defaultContainerWidth it renders the image inside
```
<PhotoAlbum
  ...
  defaultContainerWidth={300}
/>
```

What I tried to solve:
I redered the component with below check, if mounted===true then only the component will render, esle it wont. But it didnt solve either
```
const [mounted, setMounted] = useState(false);

useEffect(() => {
  setMounted(true)
}, []);
```

Any help or suggestion would be appriciated.

**Comment by igordanchenko** [comment]
Does this answer your question?

https://react-photo-album.com/#HowItWorks_SSR

**Reply by ashiq-prospero** [reply]
there's no other way other than defaultContainerWidth is that right?

**Reply by igordanchenko** [reply, marked as answer]
Well, the [documentation](https://react-photo-album.com/#HowItWorks_SSR) is pretty clear about it.

> To enable server-side rendering, be sure to specify `defaultContainerWidth` prop. Otherwise, React Photo Album produces empty markup on the server and renders on the client only after hydration.
